.github/workflows/CI.yaml

```diff
@@ -22,10 +26,14 @@ jobs:
       with:
         version: 1.3.2
 
-    - run: poetry install --with dev
+    - run: poetry install --with dev --all-extras
+
+    - name: Install pytorch CPU for ubuntu-latest
+      run: poetry run pip uninstall -y torch torchvision; poetry run pip install torch torchvision --index-url https://download.pytorch.org/whl/cpu
+      if: matrix.os == 'ubuntu-latest'
 
     - run: poetry run black --check .
 
     - run: poetry run mypy .
 
-    - run: poetry run pytest --cov
+    - run: poetry run pytest --cov
```

Comment:
I would advise generating a pytest HTML report and saving it to the CI artifacts (for 1 month).  
But that's unrelated to this PR, just a general thought 😬